### PR TITLE
feat: Array.attach

### PR DESCRIPTION
**Pull request by nomeata**
This adds `Array.attach`, an analogue to `List.attach`, to facilitate
well-founded recursion through arrays.

### CODE REVIEW

**Comment by nomeata** [comment]
This is extracted from #332, but probably useful on its own. I just told David Christiansen to use it, only to find that it was still on a PR.

I had to put it into its own file due to the dependency on `List.attach`, it seems.

**Comment by david-christiansen** [comment]
Thanks!

**Comment by nomeata** [comment]
awaiting-review

File: Std/Data/Array/Attach.lean [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import Std.Data.List.Basic
```
Comment (digama0) [inline Std/Data/Array/Attach.lean:1]: Can this go in `Std.Data.Array.Basic`? Assuming it doesn't increase the imports.

**Comment by digama0** [comment]
It seems that it did require a cross-cutting import after all. I've rejiggered things to avoid that. (Yes, some documentation regarding the allowed imports is overdue. The short version is that `Std.A.Basic` should not import `Std.B.Basic` unless it is doing so already, and requirements of std definitions within itself should be moved to a `Init` folder.)

**Comment by nomeata** [comment]
Thanks for that help!

**Pull request merged**